base/loading.jl

```diff
@@ -1246,6 +1252,13 @@ module_build_id(m::Module) = ccall(:jl_module_build_id, UInt64, (Any,), m)
 isvalid_cache_header(f::IOStream) = (0 != ccall(:jl_read_verify_header, Cint, (Ptr{Cvoid},), f.ios))
 isvalid_file_crc(f::IOStream) = (_crc32c(seekstart(f), filesize(f) - 4) == read(f, UInt32))
 
+struct CacheHeaderIncludes
```

Comment:
For the future, it would be good to include the path in which we load the package from here as well. Then issue https://github.com/JuliaLang/julia/issues/33210 could be fixed really easily. It (of course) doesn't have to be done here but I bring it up because maybe the name of the struct is a bit too narrow if we later include the load path?

Reply:
I'm not sure this is the right place, though. How about I generalize `pkgcachefiles` to be a `Dict(id => (pkgpath, cachefilepath))`?

Reply:
Sure, at some point I think I also want to put a version numer in there. Anyway, just wanted to to wanted to point out that in the future we might want even more meta data available from loading.

Reply:
This implementation should future-proof against additional info.